### PR TITLE
Ajax structure

### PR DESCRIPTION
**Pull request by MattPutnam**
I propose we do something like this. This helps organize related endpoints as the size of these objects grow larger, and reduces the number of positional arguments to something manageable.

### CODE REVIEW

**Review by panentheos — COMMENTED** [review]
Perhaps there are other ways of achieving this goal without creating coupling between the ajax calls.

For example, to reduce positional arguments, we could start using "named" arguments via an object, and include some assertions about the structure of that object to catch errors.

Also, some repeated patterns in urls could be factored out as simple helper functions.

**Comment by MattPutnam** [comment]
I don't see much cost to coupling the ajax calls. They should be pretty well set in stone.

**Review by dmohs — APPROVED** [review]
I like the mechanics of this approach. As long as the coupling doesn't cause an argument to be required where it otherwise wouldn't be, I think this is an improvement and has some concision benefit. I actually think the forced consistency makes the (unnamed) positional arguments more palatable.

**Review by panentheos — APPROVED** [review]
I'll go with the consensus here.

File: src/libs/ajax.js [hunk 1]
```diff
@@ -72,41 +72,48 @@ const fetchBuckets = (path, ...args) => fetchOk(`https://www.googleapis.com/${pa
...
-
   listNotebooks: (namespace, name) => {
     return Sam.token(namespace)
       .then(token => fetchBuckets(`storage/v1/b/${name}/o?prefix=notebooks/`, authOpts(token)))
       .then(parseJson)
       .then(({ items }) => _.filter(items, ({ name }) => name.endsWith('.ipynb')))
+  },
+
+  notebook: (namespace, bucket, name) => {
+    const bucketUrl = `storage/v1/b/${bucket}/o`
```
Comment (zarsky-broad) [inline src/libs/ajax.js:83]: Do you want to set up `oldNbName` as `nbName(name)` here?

**Review by zarsky-broad — APPROVED** [review]
I'm fundamentally in favor of conceptual grouping.

**Pull request merged**